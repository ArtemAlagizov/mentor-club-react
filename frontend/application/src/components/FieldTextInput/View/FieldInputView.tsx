import React from "react";
import { Styled } from "../Styled";
import { IFieldInputProps } from "./IFieldInputProps";

const FieldInputView: React.FC<IFieldInputProps> = ({
  labelText,
  value,
  onChange,
  isTextArea,
  numberOfLinesForTextArea
}) => {
  return (
    <>
      <Styled.Label>{labelText}</Styled.Label>
      {isTextArea && (
        <Styled.FieldTextArea
          rows={numberOfLinesForTextArea}
          value={value}
          onChange={onChange}
        />
      )}
      {!isTextArea && (
        <Styled.FieldInput type="text" value={value} onChange={onChange} />
      )}
    </>
  );
};

export default FieldInputView;
