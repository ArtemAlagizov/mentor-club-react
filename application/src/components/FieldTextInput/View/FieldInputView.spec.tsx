import React from "react";
import { shallow } from "enzyme";
import FieldInputView from "./FieldInputView";

describe("Given FieldInputView", () => {
  describe("when rendering", () => {
    it("should have FieldInputView matching snapshot", () => {
      const component = shallow(
        <FieldInputView labelText={"fakeInput"} onChange={jest.fn()} />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
