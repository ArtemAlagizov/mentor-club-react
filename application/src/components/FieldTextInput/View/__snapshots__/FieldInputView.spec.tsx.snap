// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given FieldInputView when rendering should have FieldInputView matching snapshot 1`] = `
<Fragment>
  <styled.label>
    fakeInput
  </styled.label>
  <Styled(Input)
    onChange={[MockFunction]}
    type="text"
  />
</Fragment>
`;
